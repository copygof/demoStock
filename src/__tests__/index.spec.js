import {
  tableCategory,
  tableBook,
  tableStock,
  getCategory,
  getCategoryById,
  addCategory,
  updateCategoryById,
  deleteCategoryById,
} from '../index'

describe('ทดสอบฟังก์ชัน category', () => {
  it('ควรที่จะ สามารถ getCategory ดึงข้อมูล category มาทั้งหมดได้', () => {
    expect(getCategory()).toMatchSnapshot()
  })

  it('ควรที่จะ สามารถ getCategoryById ดึงข้อมูล category ตาม id ได้', () => {
    expect(getCategoryById('category:001')).toMatchSnapshot()
    expect(getCategoryById('category:002')).toMatchSnapshot()
    expect(getCategoryById('category:003')).toMatchSnapshot()
    expect(getCategoryById('category:004')).toMatchSnapshot()
  })

  it('ควรที่จะ สามารถ addCategory เพิ่มข้อมูล category ใหม่ได้', () => {
    expect(addCategory('คอมพิวเตอร์')).toMatchSnapshot()
  })

  it('ควรที่จะ สามารถ addCategory เพิ่มข้อมูล category ใหม่ได้', () => {
    expect(addCategory('คอมพิวเตอร์')).toMatchSnapshot()
  })

  it('ควรที่จะ สามารถ updateCategoryById อัพเดพข้อมูล category ตาม id ได้', () => {
    expect(updateCategoryById('หมวดหมู่หนังสือเรียน', 'category:001')).toMatchSnapshot()
    expect(updateCategoryById('หมวดหมู่หนังสือนิยาย', 'category:002')).toMatchSnapshot()
    expect(updateCategoryById('หมวดหมู่หนังสือบันเทิง', 'category:003')).toMatchSnapshot()
    expect(updateCategoryById('หมวดหมู่หนังสือการเงิน', 'category:004')).toMatchSnapshot()
  })

  it('ควรที่จะ สามารถ deleteCategoryById ลบข้อมูล category ตาม id ได้', () => {
    expect(deleteCategoryById('category:001')).toMatchSnapshot()
    expect(deleteCategoryById('category:002')).toMatchSnapshot()
    expect(deleteCategoryById('category:003')).toMatchSnapshot()
    expect(deleteCategoryById('category:004')).toMatchSnapshot()
  })
})
