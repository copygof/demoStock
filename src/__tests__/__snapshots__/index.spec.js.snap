// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ทดสอบฟังก์ชัน category ควรที่จะ สามารถ addCategory เพิ่มข้อมูล category ใหม่ได้ 1`] = `
Array [
  Object {
    "_id": "category:001",
    "category": "หนังสือเรียน",
  },
  Object {
    "_id": "category:002",
    "category": "หนังสือนิยาย",
  },
  Object {
    "_id": "category:003",
    "category": "หนังสือบันเทิง",
  },
  Object {
    "_id": "category:004",
    "category": "หนังสือการเงิน",
  },
  Object {
    "_id": "category:005",
    "category": "คอมพิวเตอร์",
  },
]
`;

exports[`ทดสอบฟังก์ชัน category ควรที่จะ สามารถ addCategory เพิ่มข้อมูล category ใหม่ได้ 2`] = `
Array [
  Object {
    "_id": "category:001",
    "category": "หนังสือเรียน",
  },
  Object {
    "_id": "category:002",
    "category": "หนังสือนิยาย",
  },
  Object {
    "_id": "category:003",
    "category": "หนังสือบันเทิง",
  },
  Object {
    "_id": "category:004",
    "category": "หนังสือการเงิน",
  },
  Object {
    "_id": "category:005",
    "category": "คอมพิวเตอร์",
  },
  Object {
    "_id": "category:006",
    "category": "คอมพิวเตอร์",
  },
]
`;

exports[`ทดสอบฟังก์ชัน category ควรที่จะ สามารถ deleteCategoryById ลบข้อมูล category ตาม id ได้ 1`] = `
Array [
  Object {
    "_id": "category:002",
    "category": "หนังสือนิยาย",
  },
  Object {
    "_id": "category:003",
    "category": "หนังสือบันเทิง",
  },
  Object {
    "_id": "category:004",
    "category": "หนังสือการเงิน",
  },
  Object {
    "_id": "category:005",
    "category": "คอมพิวเตอร์",
  },
  Object {
    "_id": "category:006",
    "category": "คอมพิวเตอร์",
  },
]
`;

exports[`ทดสอบฟังก์ชัน category ควรที่จะ สามารถ deleteCategoryById ลบข้อมูล category ตาม id ได้ 2`] = `
Array [
  Object {
    "_id": "category:001",
    "category": "หนังสือเรียน",
  },
  Object {
    "_id": "category:003",
    "category": "หนังสือบันเทิง",
  },
  Object {
    "_id": "category:004",
    "category": "หนังสือการเงิน",
  },
  Object {
    "_id": "category:005",
    "category": "คอมพิวเตอร์",
  },
  Object {
    "_id": "category:006",
    "category": "คอมพิวเตอร์",
  },
]
`;

exports[`ทดสอบฟังก์ชัน category ควรที่จะ สามารถ deleteCategoryById ลบข้อมูล category ตาม id ได้ 3`] = `
Array [
  Object {
    "_id": "category:001",
    "category": "หนังสือเรียน",
  },
  Object {
    "_id": "category:002",
    "category": "หนังสือนิยาย",
  },
  Object {
    "_id": "category:004",
    "category": "หนังสือการเงิน",
  },
  Object {
    "_id": "category:005",
    "category": "คอมพิวเตอร์",
  },
  Object {
    "_id": "category:006",
    "category": "คอมพิวเตอร์",
  },
]
`;

exports[`ทดสอบฟังก์ชัน category ควรที่จะ สามารถ deleteCategoryById ลบข้อมูล category ตาม id ได้ 4`] = `
Array [
  Object {
    "_id": "category:001",
    "category": "หนังสือเรียน",
  },
  Object {
    "_id": "category:002",
    "category": "หนังสือนิยาย",
  },
  Object {
    "_id": "category:003",
    "category": "หนังสือบันเทิง",
  },
  Object {
    "_id": "category:005",
    "category": "คอมพิวเตอร์",
  },
  Object {
    "_id": "category:006",
    "category": "คอมพิวเตอร์",
  },
]
`;

exports[`ทดสอบฟังก์ชัน category ควรที่จะ สามารถ getCategory ดึงข้อมูล category มาทั้งหมดได้ 1`] = `
Array [
  Object {
    "_id": "category:001",
    "category": "หนังสือเรียน",
  },
  Object {
    "_id": "category:002",
    "category": "หนังสือนิยาย",
  },
  Object {
    "_id": "category:003",
    "category": "หนังสือบันเทิง",
  },
  Object {
    "_id": "category:004",
    "category": "หนังสือการเงิน",
  },
]
`;

exports[`ทดสอบฟังก์ชัน category ควรที่จะ สามารถ getCategoryById ดึงข้อมูล category ตาม id ได้ 1`] = `
Object {
  "_id": "category:001",
  "category": "หนังสือเรียน",
}
`;

exports[`ทดสอบฟังก์ชัน category ควรที่จะ สามารถ getCategoryById ดึงข้อมูล category ตาม id ได้ 2`] = `
Object {
  "_id": "category:002",
  "category": "หนังสือนิยาย",
}
`;

exports[`ทดสอบฟังก์ชัน category ควรที่จะ สามารถ getCategoryById ดึงข้อมูล category ตาม id ได้ 3`] = `
Object {
  "_id": "category:003",
  "category": "หนังสือบันเทิง",
}
`;

exports[`ทดสอบฟังก์ชัน category ควรที่จะ สามารถ getCategoryById ดึงข้อมูล category ตาม id ได้ 4`] = `
Object {
  "_id": "category:004",
  "category": "หนังสือการเงิน",
}
`;

exports[`ทดสอบฟังก์ชัน category ควรที่จะ สามารถ updateCategoryById อัพเดพข้อมูล category ตาม id ได้ 1`] = `
Array [
  Object {
    "_id": "category:001",
    "category": "หมวดหมู่หนังสือเรียน",
  },
  Object {
    "_id": "category:002",
    "category": "หนังสือนิยาย",
  },
  Object {
    "_id": "category:003",
    "category": "หนังสือบันเทิง",
  },
  Object {
    "_id": "category:004",
    "category": "หนังสือการเงิน",
  },
  Object {
    "_id": "category:005",
    "category": "คอมพิวเตอร์",
  },
  Object {
    "_id": "category:006",
    "category": "คอมพิวเตอร์",
  },
]
`;

exports[`ทดสอบฟังก์ชัน category ควรที่จะ สามารถ updateCategoryById อัพเดพข้อมูล category ตาม id ได้ 2`] = `
Array [
  Object {
    "_id": "category:001",
    "category": "หนังสือเรียน",
  },
  Object {
    "_id": "category:002",
    "category": "หมวดหมู่หนังสือนิยาย",
  },
  Object {
    "_id": "category:003",
    "category": "หนังสือบันเทิง",
  },
  Object {
    "_id": "category:004",
    "category": "หนังสือการเงิน",
  },
  Object {
    "_id": "category:005",
    "category": "คอมพิวเตอร์",
  },
  Object {
    "_id": "category:006",
    "category": "คอมพิวเตอร์",
  },
]
`;

exports[`ทดสอบฟังก์ชัน category ควรที่จะ สามารถ updateCategoryById อัพเดพข้อมูล category ตาม id ได้ 3`] = `
Array [
  Object {
    "_id": "category:001",
    "category": "หนังสือเรียน",
  },
  Object {
    "_id": "category:002",
    "category": "หนังสือนิยาย",
  },
  Object {
    "_id": "category:003",
    "category": "หมวดหมู่หนังสือบันเทิง",
  },
  Object {
    "_id": "category:004",
    "category": "หนังสือการเงิน",
  },
  Object {
    "_id": "category:005",
    "category": "คอมพิวเตอร์",
  },
  Object {
    "_id": "category:006",
    "category": "คอมพิวเตอร์",
  },
]
`;

exports[`ทดสอบฟังก์ชัน category ควรที่จะ สามารถ updateCategoryById อัพเดพข้อมูล category ตาม id ได้ 4`] = `
Array [
  Object {
    "_id": "category:001",
    "category": "หนังสือเรียน",
  },
  Object {
    "_id": "category:002",
    "category": "หนังสือนิยาย",
  },
  Object {
    "_id": "category:003",
    "category": "หนังสือบันเทิง",
  },
  Object {
    "_id": "category:004",
    "category": "หมวดหมู่หนังสือการเงิน",
  },
  Object {
    "_id": "category:005",
    "category": "คอมพิวเตอร์",
  },
  Object {
    "_id": "category:006",
    "category": "คอมพิวเตอร์",
  },
]
`;
